packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -236,128 +164,169 @@ fn read_file(path: &str) -> Result<String> {
     let source_code = read_file(path)?;
     let js_source_code = new_js_string(scope, &source_code);
-    let mut try_catch = v8::TryCatch::new(scope);
-    let script = v8::Script::compile(&mut try_catch, js_source_code, None).ok_or_else(|| {
-        let exception = try_catch.exception().unwrap();
-        let exception_string = exception
-            .to_string(&mut try_catch)
-            .unwrap()
-            .to_rust_string_lossy(&mut try_catch);
-
-        Error::Eval(path.into(), format!("Compile error: {exception_string}"))
-    })?;
-
-    script.run(&mut try_catch).ok_or_else(|| {
-        let exception = try_catch.exception().unwrap();
-        let exception_string = exception
-            .to_string(&mut try_catch)
-            .unwrap()
-            .to_rust_string_lossy(&mut try_catch);
-
-        Error::Eval(path.into(), format!("Execution error: {exception_string}"))
+    let mut try_catch_scope = v8::TryCatch::new(scope);
+    let script =
+        v8::Script::compile(&mut try_catch_scope, js_source_code, None).ok_or_else(|| {
+            let exception = exception_as_error(&mut try_catch_scope);
+
+            Error::Eval(path.into(), format!("Compile error: {exception}"))
+        })?;
+
+    script.run(&mut try_catch_scope).ok_or_else(|| {
+        let exception = exception_as_error(&mut try_catch_scope);
+
+        Error::Eval(path.into(), format!("Execution error: {exception}"))
     })
 }
 
-fn import_file<'s>(
-    scope: &mut v8::HandleScope<'s>,
-    context: v8::Local<'s, v8::Context>,
-    path: &str,
-    args: &[Value<'s>],
-) -> Result<Value<'s>> {
-    let v = eval_file(scope, path)?;
-    let f: Function<'_> = v.try_into().map_err(|err| {
-        Error::FileImport(
-            path.into(),
-            format!("Could not convert file into Function: {err}"),
-        )
-    })?;
+/// Cache module to not evaluate them twice
+struct ModuleMap {
+    path_to_module: HashMap<String, v8::Global<v8::Module>>,
+}
 
-    let global_context = context.global(scope);
-    let imported = call_js_function(scope, f, global_context.into(), args)
-        .map_err(|err| Error::FileImport(path.into(), format!("Could not call function: {err}")))?;
+impl ModuleMap {
+    fn new() -> Self {
+        Self {
+            path_to_module: HashMap::new(),
+        }
+    }
 
-    Ok(imported)
+    fn import_module<'s>(
+        scope: &mut v8::HandleScope<'s>,
+        path: &str,
+    ) -> Result<v8::Local<'s, v8::Module>> {
+        let module_map = scope
+            .get_slot::<Rc<RefCell<ModuleMap>>>()
+            .expect("ModuleMap is not present in isolate slots")
+            .clone();
+
+        if let Some(module) = module_map.borrow().path_to_module.get(path) {
+            return Ok(v8::Local::new(scope, module));
+        }
+
+        let source_code = read_file(path)
+            .map_err(|err| Error::AccessJavascriptImport(path.to_string(), format!("{err}")))?;
```

Comment:
We literally just fixed this in https://github.com/hashintel/hash/pull/493/commits/b6bbddae8db261d0db672ec6b393545a1fa706fa ...

```suggestion
            .map_err(|err| Error::AccessJavascriptImport(path.to_string(), err.to_string()))?;
```

Reply:
I blame VSCode sorry.
I made a few commits and had a bunch to pull so I let it handle it...